backend/src/test/java/com/fellas/bespoke/client/AnnotationServerClientTest.java

```diff
@@ -0,0 +1,99 @@
+package com.fellas.bespoke.client;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fellas.bespoke.TestUtils;
+import com.fellas.bespoke.client.response.AnnotationServerCreateResponse;
+import com.fellas.bespoke.client.response.AnnotationServerSearchResponse;
+import com.fellas.bespoke.exception.AnnotationServerException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.http.*;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.web.client.RestTemplate;
+
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AnnotationServerClientTest {
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @InjectMocks
+    private AnnotationServerClient sut;
+
+    @Before
+    public void setUp() {
+        ReflectionTestUtils.setField(sut, "annotationServerApiUrl", "http:localhost/annotation/w3c/");
+        ReflectionTestUtils.setField(sut, "annotationContainerId", "container-id");
+    }
+
+    @Test
+    public void testCreateAnnotation_Success() throws JsonProcessingException {
+        //Prepare
+        final String annotationServerRequestJson = TestUtils.createDummyAnnotationServerCreateRequestJson();
+        final HttpHeaders httpHeaders = TestUtils.createDummyAnnotationServerHttpHeaders();
+        final ResponseEntity<AnnotationServerCreateResponse> annotationServerCreateResponseResponseEntity = new ResponseEntity<>(HttpStatus.CREATED);
+        when(restTemplate.exchange("http:localhost/annotation/w3c/container-id", HttpMethod.POST,
+                new HttpEntity<>(annotationServerRequestJson,httpHeaders), AnnotationServerCreateResponse.class))
+                .thenReturn(annotationServerCreateResponseResponseEntity);
+
+        //Test
+        sut.createAnnotation(annotationServerRequestJson);
```

Comment:
Where is verification? It verifies nothing. You should at least verify an object equality, a method call or a response object.

Reply:
If you don't have any of these object, than there is something wrong in the actual implementation.